### PR TITLE
rt: rm internal Park,Unpark traits

### PR DESCRIPTION
**Pull request by carllerche**
Years ago, Tokio was organized as a cluster of separate crates. This architecture required a trait to represent "park the thread and do work.". When all crates were combined into a single crate, the `Park` and `Unpark` traits remained as internal details.

This patch removes these traits as they are no longer needed. This is in service of a future refactor that will decouple the various resource drivers and store them in a single struct instead of nesting them. However, in the meantime, removing the Park/Unpark traits adds a bit of messy code to make the conditional compilation work. This code will (hopefully) be short-lived.

I am trying to submit refactors in small, approachable chunks to avoid having to review massive PRs.

### CODE REVIEW

File: tokio/src/runtime/driver.rs [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 cfg_io_driver! {
-    type IoDriver = crate::runtime::io::Driver;
-    type IoStack = crate::park::either::Either<ProcessDriver, ParkThread>;
+    pub(crate) type IoDriver = crate::runtime::io::Driver;
     pub(crate) type IoHandle = Option<crate::runtime::io::Handle>;
 
-    fn create_io_stack(enabled: bool) -> io::Result<(IoStack, IoHandle, SignalHandle)> {
-        use crate::park::either::Either;
+    #[derive(Debug)]
+    pub(crate) enum IoStack {
```
Comment (carllerche) [inline tokio/src/runtime/driver.rs:14]: Some boilerplate / duplication here, but I aim to clean this up later.

File: tokio/src/runtime/io/mod.rs [hunk 2]
```diff
@@ -145,7 +144,35 @@ impl Driver {
         }
     }
 
-    fn turn(&mut self, max_wait: Option<Duration>) -> io::Result<()> {
+    // TODO: remove this in a later refactor
+    cfg_not_rt! {
```
Comment (carllerche) [inline tokio/src/runtime/io/mod.rs:148]: Same, will clean up later.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -281,6 +281,8 @@ jobs:
         run: cargo install cargo-hack
       - name: check --each-feature
         run: cargo hack check --all --each-feature -Z avoid-dev-deps
+      - name: check net,time
+        run: cargl check -p tokio --no-default-features --features net,time -Z avoid-dev-deps
```
Comment (Noah-Kennedy) [inline .github/workflows/ci.yml:285]: ph'nglui mglw'nafh Ferris `Cargl` wgah'nagl fhtagn

**Review by Noah-Kennedy — APPROVED** [review]
I think this is fine once you get it through CI.

**Review by hawkw — APPROVED** [review]

**Pull request merged**